index.html

```diff
@@ -618,6 +618,85 @@ <h4>
         </section>
       </section>
     </section>
+    <section>
+      <h2>
+        Automation
+      </h2>
+      <p>
+        The Device Posture API pose a challenge to test authors, as fully
+        exercising interface requires physical hardware devices. To address this
+        challenge this document defines a [[WEBDRIVER2]] [=extension commands=]
+        that allows users to control the reported device posture and simulate a
+        real device.
+      </p>
+      <h3>
+        Extension Commands
+      </h3>
+      <h4>
+        Set device posture
+      </h4>
+      <table class="simple">
+        <tr>
+          <th>
+            HTTP Method
+          </th>
+          <th>
+            [=extension command URI Template|URI Template=]
+          </th>
+        </tr>
+        <tr>
+          <td>
+            POST
+          </td>
+          <td>
+            deviceposture
+          </td>
+        </tr>
+      </table>
+      <p>
+        This [=extension commands|extension command=] changes device posture to a specific
+        {{DevicePostureType}}.
+      </p>
+      <table class="simple">
+        <caption>
+          Properties of the parameters argument used by this algorithm
+        </caption>
+        <tr>
+          <th>
+            Parameter name
+          </th>
+          <th>
+            Value type
+          </th>
+          <th>
+          Required
+          </th>
+        </tr>
+        <tr>
+          <td>
+            posture
+          </td>
+          <td>
+            String
+          </td>
+          <td>
+            yes
+          </td>
+        </tr>
+      </table>
+      <p>
+        The [=remote end steps=] are:
+      </p>
+      <ol class="algorithm" data-cite="WEBDRIVER2">
+        <li>If |posture| is not {{DevicePostureType}}, return [=error=] with
```

Comment:
See https://w3c.github.io/webdriver/#processing-model and what the Generic Sensor spec and others do. You need to fetch this variable from somewhere; at this point, your remote end steps are given "`session`, `URL variables`, and `parameters`" (step 1.6.6).

Reply:
I have all along wondered should the URI be "/session/{session id}/deviceposture" with parameters or "/session/{session id}/deviceposture/{posture}". Latter would be simpler , which is always better.

But I have wondered if single parameter ("posture") is enough for this service. This is little bit connected to [Emulation.setDeviceMetricsOverride()](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/devtools/protocol/emulation_handler.cc?q=EmulationHandler::SetDeviceMetricsOverride%20file:.cc&ss=chromium%2Fchromium%2Fsrc) as it is not happy if only posture variable is provided. This can be seen in chromedriver/window_commands.cc @ [DevicePosture_WebDriver](https://github.com/JuhaVainio/applications.browsers.thirdparty.chromium.chromium/commits/DevicePosture_WebDriver/)-branch as I have to provide also width, hight, etc.

Maybe this should be first solved before going too deep in style of text.